### PR TITLE
NSIS: Add /CONFIG and /DISABLEREPORTING flag

### LINKED ISSUE #4005: Agent Flow config file argument
### Request

When installing the Grafana Agent Flow, there should be the possibility to define a specific config.river
E.g.
`grafana-agent-flow-installer.exe /S /config="./config.river"`

### Use case

On rollouts it would be possible to define the content of the config.river directly without changing the config file after the installation.

### PR DESCRIPTION
**Pull request by jkroepke**
#### PR Description

Add /CONFIG and /DISABLEREPORTING flag to NSIS installer

#### Which issue(s) this PR fixes

Fixes #4005

#### PR Checklist

- [ ] CHANGELOG.md updated
- [x] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: packaging/grafana-agent-flow/windows/install_script.nsis [hunk 1]
```diff
@@ -105,12 +116,24 @@ Function InitializeRegistry
...
+    StrCpy $DisableReportingFlag ""
+  ${EndIf}
+
   # Define the arguments key, which holds arguments to pass to the
   # service.
   nsExec::ExecToLog 'Reg.exe query "${REGKEY}" /reg:64 /v Arguments'
   Pop $0
   ${If} $0 == 1
-    nsExec::ExecToLog 'Reg.exe add "${REGKEY}" /reg:64 /v Arguments /t REG_MULTI_SZ /d "run"\0"$INSTDIR\config.river"\0"--storage.path=$APPDATA\${APPNAME}\data"\0'
+    nsExec::ExecToLog 'Reg.exe add "${REGKEY}" /reg:64 /v Arguments /t REG_MULTI_SZ /d "run"\0"$ConfigFlag"\0"--storage.path=$APPDATA\${APPNAME}\data"\0"$DisableReportingFlag"'
```
Comment (mattdurham) [inline packaging/grafana-agent-flow/windows/install_script.nsis:136]: For some reason the quotes look off, though I assume you have tested the scenarios?

File: docs/sources/flow/setup/install/windows.md [hunk 2]
```diff
@@ -51,6 +51,11 @@ To do a silent install of Grafana Agent on Windows, perform the following steps.
 
    Replace `PATH_TO_INSTALLER` with the path where the unzipped installer executable is located.
 
+### Silent options
+
+* `/CONFIG=<path>` Path to config file. Default: `$INSTDIR\config.river`
+* `/DISABLEREPORTING=<yes|no>` Disable Analytics of Agent. Default: `no`
```
Comment (clayton-cornell) [inline docs/sources/flow/setup/install/windows.md:57]:
```suggestion
### Silent install options

* `/CONFIG=<path>` Path to the configuration file. Default: `$INSTDIR\config.river`
* `/DISABLEREPORTING=<yes|no>` Disable [data collection][]. Default: `no`
```

We could add a link to the data collection topic. We need to add the following in the `doc/reference` shortcode block (somewhere between lines 72 and 77)

```
[data collection]: "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/data-collection.md"
[data collection]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/monitor-infrastructure/agent/data-collection.md"
```

**Review by mattdurham — APPROVED** [review]
LGTM

**Comment by mattdurham** [comment]
Will merge once Claytons feedback is resolved.

File: CHANGELOG.md [hunk 3]
```diff
@@ -54,6 +54,8 @@ Main (unreleased)
 
 - Added support for python profiling to `pyroscope.ebpf` component. (@korniltsev)
  
+- NSIS: Add /CONFIG and /DISABLEREPORTING flag (@jkroepke)
```
Comment (mattdurham) [inline CHANGELOG.md:57]: nit: NSIS -> Windows Flow Installer, I imagine most people arent familiar with NSIS

**Pull request merged**